server/controllers/finance/cash.js

```diff
@@ -0,0 +1,247 @@
+/**
+* Cash Controller
+*
+* This controller is responsible for processing cash payments for patients. The
+* payments can either be against an previous invoice (sale payment) or a future
+* invoice (cautionary payment).
+*
+* In order to reduce the burden of accounting on the user, the user will first
+* select a cashbox which implicitly bundles in cash accounts for all supported
+* currencies.  The API accepts a cashbox ID during cash payment creation and
+* looks up the correct account based on the cashbox_id + currency.
+*/
+var db   = require('../../lib/db');
+var uuid = require('../../lib/guid');
+
+// check if an object is empty
+// TODO - can this be from a library?
+function empty(object) {
```

Comment:
I do not say, We must no have a if instruction at the server side. The Data Access Layer is not responsible of validation, the database him self is responsible of data integrity witch is the ACID properties  (Atomicity, Coherence, Isolation, Durability).

The DAL or DAO must be independent from business layer, doing validation in DAL is to encrust some business logic in it so the DAO will be dependent to the Business logic layer, the problem is when the enterprise will decide to change some business rule, we will be supposed to touch the Business Layer (which is essentially our client side) and The DAL, that situation is not good. Isolate DAL from BL (Business Layer) is very important.

If possible we can make our database robust by making a good implementation it is one of our goal I think.

@jniles @sfount